### ISSUE TITLE
Improve message if StringSubject expected/actual is an empty string

### ISSUE THREAD

**Issue by groakley** [google/truth#623]
If comparing against an empty string, the failure message is a bit difficult to understand due to the string representation of an empty string being, well, empty. It could be made more understandable by special-casing empty strings (or perhaps all whitespace-only strings):

Current:
```
expected: ABC
but was : 
	at com.google.common.truth.StringSubjectTest.stringEqualityEmptyToNonEmptyFail(StringSubjectTest.java:150)
...
```

Proposal:
```
expected: ABC
but was an empty string 
	at com.google.common.truth.StringSubjectTest.stringEqualityEmptyToNonEmptyFail(StringSubjectTest.java:150)
...
```

**Comment by kluever** [comment]
Sounds pretty reasonable.

**Comment by cpovirk** [comment]
+1. We could stick a special case right around here:

https://github.com/google/truth/blob/7a58a45ba139f0b7e56bc2ffbf7733fa5351666c/core/src/main/java/com/google/common/truth/Subject.java#L863

Maybe it should apply to the null cases, too, and even to the `==`-comparison case. Possibly it would make sense to instead do it near where we do the recently added special handling for whitespace:

https://github.com/google/truth/blob/7a58a45ba139f0b7e56bc2ffbf7733fa5351666c/core/src/main/java/com/google/common/truth/Subject.java#L839

Probably we'd do this for strings only (i.e., not for other objects whose `toString()` might be empty), again similar to what we did for whitespace:

https://github.com/google/truth/blob/7a58a45ba139f0b7e56bc2ffbf7733fa5351666c/core/src/main/java/com/google/common/truth/Subject.java#L879-L881

**Comment by groakley** [comment]
Great, I will send a PR

**Comment by groakley** [comment]
PR #624

**Comment by groakley** [comment]
Fixed by #626
